roku/device/roku-tv.groovy

```diff
@@ -615,15 +625,23 @@ String translateAppToInput(appName) {
 }
 
 void setCurrentApplication(currentApp) {
+    if (this[SETTING_LOG_ENABLE]) { log.info "setCurrentApplication: Executed " }
     def previousApp = device.latestValue('application')
```

Comment:
here, we should use `device.currentValue('application', true)` to force it to skip the cache.  This is likely the source of the issue you are seeing.

Reply:
This may be a issue, but i don't think it was causing a issue with the following steps in the module. You can see with the additional logging that the comparison works. On line 633 this appears to show the correct values for the comparison. The 2 second delay on line 642 seems to provide enough time though for the schedule to actually work. I am not clear why the delay is needed, but that does seem to resolve the problem with the Query Media Player not scheduling. Without that 2 second delay it was sporadic on if it would schedule or not.

Reply:
I need to see if I can get this to reproduce with some level of repeatability.  Seeing all the events and logging first hand is tremendous help in isolating the source of a race condition, which is what this issue sounds like.